File: app/src/main/java/com/example/daong/activitykotlin/Main2Activity.java
package com.example.daong.activitykotlin;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Gravity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.miguelcatalan.materialsearchview.MaterialSearchView;

public class Main2Activity extends AppCompatActivity {

    final String TAG = "States";
    MaterialSearchView materialSearchView;
    String[] list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        Intent intent = getIntent();
        String text = intent.getStringExtra(MainActivity.EXTRA_TEXT);

        TextView textView = (TextView) findViewById(R.id.textview);

        textView.setText(text);

        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle("Message");
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),MainActivity.class));
                finish();
            }
        });

        list = new String[]{"AD340", "Android", "Mobile App", "Google", "Programming", "Android Developer"};
        materialSearchView = (MaterialSearchView) findViewById(R.id.mysearch);
        materialSearchView.setSuggestions(list);
        materialSearchView.setEllipsize(true);
        materialSearchView.setOnQueryTextListener(new MaterialSearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                //Here create your filtering
                Toast.makeText(getApplicationContext(), query, Toast.LENGTH_SHORT).show();
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                //change if typing
                return false;
            }
        });

        materialSearchView.setOnSearchViewListener(new MaterialSearchView.SearchViewListener() {
            @Override
            public void onSearchViewShown() {

            }

            @Override
            public void onSearchViewClosed() {

            }
        });

        Log.d(TAG, "Main2Activity: onCreate()");
    }

    @Override
    public void onBackPressed() {

        if (materialSearchView.isSearchOpen()) {
            materialSearchView.closeSearch();
        }else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.setting_option, menu);
        MenuItem item = menu.findItem(R.id.search);
        materialSearchView.setMenuItem(item);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_setting) {
            Toast toast = Toast.makeText(Main2Activity.this, "This is setting option menu!", Toast.LENGTH_SHORT);
            toast.setGravity(Gravity.BOTTOM | Gravity.CENTER_HORIZONTAL, 0, 10);
            toast.show();
            return true;
        }
        if(id == R.id.search) {
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }

    @Override
    protected void  onRestart() {
        super.onRestart();
        Log.d(TAG, "Main2Activity: onRestart()");
    }

    @Override
    protected void  onStart() {
        super.onStart();
        Log.d(TAG, "Main2Activity: onStart()");
    }

    @Override
    protected void  onResume() {
        super.onResume();
        Log.d(TAG, "Main2Activity: onResume()");
    }

    @Override
    protected void  onPause() {
        super.onPause();
        Log.d(TAG, "Main2Activity: onPause()");
    }

    @Override
    protected void  onStop() {
        super.onStop();
        Log.d(TAG, "Main2Activity: onStop()");
    }

    @Override
    protected void  onDestroy() {
        super.onDestroy();
        Log.d(TAG, "Main2Activity: onDestroy()");
    }
}

File: app/src/main/java/com/example/daong/activitykotlin/Main2Activity.kt
package com.example.daong.activitykotlin

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.util.Log
import android.view.Gravity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.example.daong.activitykotlin.MainActivity.EXTRA_TEXT

class Main2Activity : AppCompatActivity() {

    internal val TAG = "States"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        toolbar.title = "Message"
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)

        toolbar.setNavigationOnClickListener {
            startActivity(Intent(applicationContext, MainActivity::class.java))
            finish()
        }

        val intent = intent
        val text = intent.getStringExtra(EXTRA_TEXT)

        val textView = findViewById(R.id.textview) as TextView

        textView.text = text
        Log.d(TAG, "Main2Activity: onCreate()")
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.setting_option, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.action_setting) {
            val toast = Toast.makeText(this@Main2Activity, "This is setting option menu!", Toast.LENGTH_SHORT)
            toast.setGravity(Gravity.BOTTOM or Gravity.CENTER_HORIZONTAL, 0, 10)
            toast.show()
            return true
        } else {
            return super.onOptionsItemSelected(item)
        }
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "Main2Activity: onRestart()")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "Main2Activity: onStart()")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "Main2Activity: onResume()")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "Main2Activity: onPause()")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "Main2Activity: onStop()")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "Main2Activity: onDestroy()")
    }
}
